DEVDOCS-2077: Customizing JavaScript (#616)
bigcommerce/dev-docs: docs/stencil-docs/javascript-and-event-hooks/customizing-javascript.md

<div class="otp" id="no-index">

### On this page
- [Modifying a page's Javascript](#modifying-a-pages-javascript)

> - [Bringing in handlebars context](#bringing-in-handlebars-context)

Reviewer: ```suggestion
- [Bringing in Handlebars context](#bringing-in-handlebars-context)

```

- [Installing libraries](#installing-libraries)

</div>

Most [Cornerstone theme](https://github.com/bigcommerce/cornerstone) page template files located in `templates/pages/` have a corresponding `.js` file in `assets/js/theme/`. These JavaScript files contain event handlers and logic for managing page specific elements and actions.

> For example, `assets/js/theme/product.js` corresponds to `templates/pages/product.html` and contains a `productReviewHandler()` and `bulkPricingHandler()`; these functions are implemented within a derived `Product` class which extends an abstract class called `PageManager` (this same pattern is repeated in all `assets/js/theme/*.js` files).

Reviewer: ```suggestion
For example, `assets/js/theme/product.js` corresponds to `templates/pages/product.html` and contains a `productReviewHandler()` and `bulkPricingHandler()`. These functions are implemented within a derived `Product` class which extends an abstract class called `PageManager`. This same pattern is repeated in all `assets/js/theme/*.js` files.

```

> By default, all derived `PageManager` classes contain an `onReady` method functionally similar to `JQuery.ready()`. Additionally, certain pages have unique event handlers. For example the `cartUpdate` handler in `assets/js/cart.js` runs each time certain cart elements are changed. Developers can customize and enhance page behavior and functionality by editing code within these event handlers (or by creating their own event handlers).

Reviewer: ```suggestion
By default, all derived `PageManager` classes contain an `onReady` method functionally similar to `JQuery.ready()`. Additionally, certain pages have unique event handlers. For example, the `cartUpdate` handler in `assets/js/cart.js` runs each time certain cart elements are changed. Developers can customize and enhance page behavior and functionality by editing code within these event handlers or by creating their own event handlers.

```

To demonstrate, this article describes how to add some very simple JavaScript to `product.js`.

## Modifying a page's Javascript

> In this example, we will add some "hello world" text to the product page's `onReady` event. We'll first add a call to the function; then, we will implement the function itself.

Reviewer: ```suggestion
In this example, we will add some "Hello World!" text to the product page's `onReady` event. We'll first add a call to the function. Then, we will implement the function itself.

```

> In `assets/js/theme/product.js` file, add a call to `this.helloWorld()` to the bottom of the `onReady()` method (implemented in next step).

Reviewer: ```suggestion
In `assets/js/theme/product.js` file, add a call to `this.helloWorld()` to the bottom of the `onReady()` method, which we'll implement in the next step.

```

}
```

> In your browser, refresh any product-details pop-up or page to see your new `"Hello World"` message. If you are using the Stencil CLI and browsing to localhost, you may need to restart for the most recent changes to be reflected.

Reviewer: ```suggestion
In your browser, refresh any product details pop-up or page to see your new "Hello World" message. If you are using the Stencil CLI and browsing to localhost, you may need to restart for the most recent changes to be reflected.

```

> ## Bringing in handlebars context

Reviewer: ```suggestion
## Bringing in Handlebars context

```

}
```

> Note that the `console.log(this.context.themeImageSizes);` statement will report the configured image size, following the `"Hello World"` message previously defined in the [Customizing JavaScript for Individual Pages](#customizing-javascript_customizing) above.

Reviewer: ```suggestion
Note that the `console.log(this.context.themeImageSizes);` statement will report the configured image size, following the "Hello World!" message previously defined in the [Customizing JavaScript for Individual Pages](#customizing-javascript_customizing) above.

```

```
[/assets/js/theme/product.js]: Hello World!
product.js:63 [/assets/js/theme/product.js]: 100x100
```

## Installing libraries

> Once you have installed the default Stencil theme, you can add custom JavaScript libraries with `npm`.

Reviewer: ```suggestion
Once you have installed the default Stencil theme, you can add custom JavaScript libraries with npm.

```
